portal-frontend/src/app/shared/owner-dialogs/parcel-owners/parcel-owners.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { MatTableDataSource } from '@angular/material/table';
+import { ApplicationDocumentService } from '../../../services/application-document/application-document.service';
+import { ApplicationOwnerDto } from '../../../services/application-owner/application-owner.dto';
+import { ApplicationOwnerService } from '../../../services/application-owner/application-owner.service';
+import { NoticeOfIntentDocumentService } from '../../../services/notice-of-intent-document/notice-of-intent-document.service';
+import { NoticeOfIntentOwnerDto } from '../../../services/notice-of-intent-owner/notice-of-intent-owner.dto';
+import { NoticeOfIntentOwnerService } from '../../../services/notice-of-intent-owner/notice-of-intent-owner.service';
+import { ConfirmationDialogService } from '../../confirmation-dialog/confirmation-dialog.service';
+import { OWNER_TYPE } from '../../dto/owner.dto';
+import { CrownOwnerDialogComponent } from '../crown-owner-dialog/crown-owner-dialog.component';
+import { OwnerDialogComponent } from '../owner-dialog/owner-dialog.component';
+
+@Component({
+  selector: 'app-parcel-owners[owners][fileId][submissionUuid][ownerService]',
+  templateUrl: './parcel-owners.component.html',
+  styleUrls: ['./parcel-owners.component.scss'],
+})
+export class ParcelOwnersComponent {
+  @Output() onOwnersUpdated = new EventEmitter<void>();
+  @Output() onOwnerRemoved = new EventEmitter<string>();
+
+  @Input() ownerService!: ApplicationOwnerService | NoticeOfIntentOwnerService;
+  @Input() documentService!: ApplicationDocumentService | NoticeOfIntentDocumentService;
+  dataSource = new MatTableDataSource([]);
+
+  @Input()
+  public set owners(owners: ApplicationOwnerDto[] | NoticeOfIntentOwnerDto[]) {
+    const sorted = owners.sort(this.sortOwners);
+    // @ts-ignore
+    this.dataSource = new MatTableDataSource(sorted);
```

Comment:
I wonder what was the message here that you needed to use ts-ignore

Reply:
Something along the lines of MatTableSource<ApplicationOwnerDto[] | NoticeOfIntentOwnerDto[]> is not assignable to CdkMatTableDataSource<ApplicationOwnerDto>, for some reason the table data source hated getting passed an array of union types.